src/main/java/core/basesyntax/operation/SupplyHandler.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax.operation;
+
+import core.basesyntax.service.ShopService;
+import core.basesyntax.transaction.FruitTransaction;
+
+public class SupplyHandler implements OperationHandler {
+    private ShopService shopService;
+
+    public SupplyHandler(ShopService shopService) {
+        this.shopService = shopService;
+    }
+
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (transaction.getOperation() == FruitTransaction.Operation.SUPPLY) {

```

Comment:
As you're using Strategy pattern, you don't need to check operation type in the apply() method. Each Strategy implementation should already correspond to a specific operation. So, you can remove this check.